Expensify/App
# Fix: navigation to report screen from the SideModals

### Details


https://github.com/Expensify/App/issues/4756#issuecomment-960645754
It fixes various issues
- Preserving the proper history
- Clearing the old Draft message from the composer


### Fixed Issues

$ https://github.com/Expensify/App/issues/4756

### Tests | QA Steps
1. Open the App on any device.
2. Go to any report.
3. type a draft message.
4. Switch the report from SearchPage, you should not see the draft comment from the previous report.

---
1. Open the App on any device.
2. Go to any report.
3. Try switching reports from LHN, search page, etc.
4. You should be able to switch reports.


### Tested On


### Screenshots


#### Web | Desktop


https://user-images.githubusercontent.com/24370807/140308514-889cfe53-bb58-40f3-a2eb-d62b7eeeaf5e.mp4



https://user-images.githubusercontent.com/24370807/140313360-4c8639a9-1ddc-46df-bbd9-9d0ef14a11fc.mp4




#### Mobile Web


https://user-images.githubusercontent.com/24370807/140308654-1d64b44c-551c-435c-a68f-5f445b219645.mp4

#### iOS


#### Android

https://user-images.githubusercontent.com/24370807/140309164-75f12256-11bd-4b69-bbf1-920bd7bb2989.mp4

https://user-images.githubusercontent.com/24370807/140313295-1f1fc149-d779-4ffc-9926-47d7b0b8c636.mp4
